types/ramda/test/pick-tests.ts

```diff
@@ -3,38 +3,36 @@ import * as R from "ramda";
 (() => {
     const abcdObj: { a: 1; b: 2; c: 3; d: 4 } = { a: 1, b: 2, c: 3, d: 4 };
 
-    const a10: { a: number; d: number } = R.pick(["a", "d"], { a: 1, b: 2, c: 3, d: 4 }); // => {a: 1, d: 4}
-    const a11: { a: 1; d: 4 } = R.pick(["a", "d"], abcdObj); // => {a: 1, d: 4}
+    const a10: { a: number; d: number } = R.pick(["a", "d"], { a: 1, b: 2, c: 3, d: 4 }); // {a: 1, d: 4}
+    const a11: { a: 1; d: 4 } = R.pick(["a", "d"], abcdObj); // {a: 1, d: 4}
 
     // @ts-expect-error
-    const a20: { a: number } = R.pick(["a", "e", "f"], { a: 1, b: 2, c: 3, d: 4 }); // => {a: 1}
+    const a20: { a: number } = R.pick(["a", "e", "f"], { a: 1, b: 2, c: 3, d: 4 }); // {a: 1}
     // @ts-expect-error
-    const a21: { a: 1 } = R.pick(["a", "e", "f"], abcdObj); // => {a: 1}
+    const a21: { a: 1 } = R.pick(["a", "e", "f"], abcdObj); // {a: 1}
 
     // @ts-expect-error
-    const a30: { a: number } = R.pick(["a", "e", "f"])({ a: 1, b: 2, c: 3, d: 4 }); // => {a: 1}
+    const a30: { a: number } = R.pick(["a", "e", "f"])({ a: 1, b: 2, c: 3, d: 4 }); // {a: 1}
     // @ts-expect-error
-    const a31: { a: 1 } = R.pick(["a", "e", "f"])(abcdObj); // => {a: 1}
+    const a31: { a: 1 } = R.pick(["a", "e", "f"])(abcdObj); // {a: 1}
 
     // @ts-expect-error
-    const a40: {} = R.pick(["a", "e", "f"], [1, 2, 3, 4]); // => {}
+    const a40: {} = R.pick(["a", "e", "f"], [1, 2, 3, 4]); // {}
 
     const abcdTuple: ["a", "b", "c", "d"] = ["a", "b", "c", "d"];
-    const a50: { 1: string; 2: string } = R.pick([1, 2], ["a", "b", "c", "d"]); // => {1: 'b', 2: 'c'}
-    const a51: { 1: "b"; 2: "c" } = R.pick([1, 2], abcdTuple); // => {1: 'b', 2: 'c'}
-    const a52: { 0: string } = R.pick([12, 0], ["a", "b", "c", "d"]); // => {0: 'a'}
-    const a53: { 0: "a" } = R.pick([12, 0], abcdTuple); // => {0: 'a'}
+    const a50: Pick<string[], 1 | 2> = R.pick([1, 2], ["a", "b", "c", "d"]); // {1: 'b', 2: 'c'}
```

Comment:
I was looking into this while looking into DT test failures in TS5.5. A modified version of `pick` seems to give better results:

```ts
export function pick<const Keys extends PropertyKey>(names: readonly Keys[]): <const U extends Partial<Record<Keys, unknown>>>(obj: U) => Pick<U, Keys>;
export function pick<const U, const Keys extends keyof U>(names: readonly Keys[], obj: U): Pick<U, Keys>;
```

The current types for `pick` seem terribly complicated. I'm curious what value you get from 

```ts
export function pick<const Keys extends readonly PropertyKey[]>(names: Keys): <U extends Partial<Record<ElementOf<Keys>, any>>>(obj: ElementOf<Keys> extends keyof U ? U : never) => ElementOf<Keys> extends keyof U ? Pick<U, ElementOf<Keys>> : never;
```

that you can't get from the code above?

With the simpler version, I get correct results for the tests for `a30`, `a31`, and `a60` and can remove the `// @ts-expect-error`.